src/garage/np/algos/off_policy_rl_algorithm.py

```diff
@@ -169,3 +181,48 @@ def train_once(self, itr, paths):
 
         """
         raise NotImplementedError
+
+    def _obtain_evaluation_samples(self, itr, batch_size=1000):
+        r"""Sample the policy for 10 trajectories and return average values.
+
+        Args:
+            itr (int): Number of iteration.
+            batch_size (int): Number of environment steps in one batch.
+
+        Returns:
+            dict: Evaluation trajectories, representing the best
+                current performance of the algorithm, with key
+                * itr (int): Iteration number.
+                * num_trajs (int): Number of trajectories.
+                * episode_reward_mean (float): Average episode reward.
+                * average_discounted_return (float): Average discounted return.
+                * undiscounted_returns (numpy.ndarray): A numpy array of shape
+                    :math:`(N \bullet [T])` containing the undiscounted returns
+                    for all time steps in this batch.
+
+        """
+        trajectories_num = 10
+        paths = []
+
+        for _ in range(trajectories_num):
+            path = rollout(self.env,
+                           self.policy,
+                           max_path_length=batch_size,
+                           deterministic=True)
+
+            path['returns'] = tensor_utils.discount_cumsum(
+                path['rewards'], self.discount)
+
+            paths.append(path)
+
+        average_discounted_return = (np.mean(
+            [path['returns'][0] for path in paths]))
+
+        undiscounted_returns = [sum(path['rewards']) for path in paths]
+        self.episode_reward_mean.extend(undiscounted_returns)
+
+        return dict(itr=itr,
```

Comment:
the intention of this interface is for you to provide `evaluate_performance` with the raw data and it calculates all these quantities.